pkg/tests/tasks/traffic/egress/egress_gateways_tls_sds_test.go

```diff
@@ -15,110 +15,81 @@
-			nginxServerCACert)
-
-		log.Log.Info("Configure MTLS origination for egress traffic")
-		util.KubeApplyContents("bookinfo", util.RunTemplate(EgressGatewaySDSTemplate, smcp))
-		util.KubeApplyContents(meshNamespace, meshExternalServiceEntry)
-		util.KubeApplyContents(meshNamespace, OriginateSDS)
-		time.Sleep(time.Duration(10) * time.Second)
-
-		log.Log.Info("Verify NGINX server")
-		cmd := fmt.Sprintf(`curl -sS http://my-nginx.mesh-external.svc.cluster.local`)
-		msg, err := util.PodExec("bookinfo", sleepPod, "sleep", cmd, true)
-		util.Inspect(err, "failed to get response", "", t)
-		if !strings.Contains(msg, "Welcome to nginx") {
-			t.Errorf("Expected Welcome to nginx; Got unexpected response: %s", msg)
-			log.Log.Errorf("Expected Welcome to nginx; Got unexpected response: %s", msg)
-		} else {
-			log.Log.Infof("Success. Get expected response: %s", msg)
-		}
+	NewTest(t).Id("T15").Groups(Full, InterOp).Run(func(t TestHelper) {
+		hack.DisableLogrusForThisTest(t)
+
+		ns := "bookinfo"
+		t.Cleanup(func() {
+			oc.RecreateNamespace(t, ns)
+		})
+
+		app.InstallAndWaitReady(t, app.Sleep(ns))
+
+		t.NewSubTest("ServiceEntry").Run(func(t TestHelper) {
+			t.LogStep("Perform TLS origination with an egress gateway")
+			oc.ApplyString(t, ns, ExServiceEntry)
+			t.Cleanup(func() {
+				oc.DeleteFromString(t, ns, ExServiceEntry)
+			})
+			execInSleepPod(t, ns,
+				`curl -sSL -o /dev/null -D - http://istio.io`,
+				assert.OutputContains(
+					"301",
+					"Expected 301 Moved Permanently",
+					"Unexpected response, expected 301 Moved Permanently"))
+
+			t.LogStep("Create a Gateway to external istio.io")
+			oc.ApplyTemplate(t, ns, ExGatewayTLSFileTemplate, smcp)
+			t.Cleanup(func() {
+				oc.DeleteFromTemplate(t, ns, ExGatewayTLSFileTemplate, smcp)
+			})
+			execInSleepPod(t, ns,
+				`curl -sSL -o /dev/null -D - http://istio.io`,
+				assert.OutputContains(
+					"HTTP/1.1 200 OK",
+					"Expected 200 from istio.io",
+					"Unexpected response, expected 200"))
+		})
+
+		t.NewSubTest("Gateway").Run(func(t TestHelper) {
+			t.Log("Perform mTLS origination with an egress gateway")
+			nsNginx := "mesh-external"
+			t.Cleanup(func() {
+				oc.DeleteNamespace(t, nsNginx)
+				oc.DeleteSecret(t, meshNamespace, "client-credential")
+				oc.DeleteFromTemplate(t, ns, EgressGatewaySDSTemplate, smcp)
+				oc.DeleteFromString(t, meshNamespace, meshExternalServiceEntry, OriginateSDS)
+			})
+
+			t.LogStep("Deploy nginx mTLS server and create secrets in the mesh namespace")
+
+			app.InstallAndWaitReady(t, app.NginxWithMTLS(nsNginx))
+
+			oc.CreateGenericSecretFromFiles(t, meshNamespace, "client-credential",
+				"tls.key="+nginxClientCertKey,
+				"tls.crt="+nginxClientCert,
+				"ca.crt="+nginxServerCACert)
+			oc.ApplyTemplate(t, ns, EgressGatewaySDSTemplate, smcp)
+			oc.ApplyString(t, meshNamespace, meshExternalServiceEntry, OriginateSDS)
+
+			t.Log("Send HTTP request to my-nginx in mesh-external namespace to verify the nginx server")
```

Comment:
But this isn't verifying the nginx server. This is verifying something about traffic. I'm not sure why, because it's not immediately evident from the test. That's why I think the log messages should clearly explain it.